### PR TITLE
ci: release to modal main env on push to main

### PR DESCRIPTION
**Pull request by sambarnes**
## Details

releases to prod env on push to main

just tested this in dev locally w/ act:

image

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/OpenRouterTeam/openrouter-runner/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I agree to license this contribution under the MIT LICENSE
- [x] I checked the [current PR](https://github.com/OpenRouterTeam/openrouter-runner/pulls) for duplication.

### CODE REVIEW

File: .github/workflows/release-prod.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+name: Release (prod)
+
+on:
+  push:
+    branches: [main]
+    paths: ['modal/**']
```
Comment (sambarnes) [inline .github/workflows/release-prod.yml:6]: Any other paths we need to watch?

File: .github/workflows/release-prod.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  release:
+    name: Release (prod)
+    runs-on: ubuntu-latest
+    env:
+      MODAL_TOKEN_ID: ${{ secrets.MODAL_TOKEN_ID }}
+      MODAL_TOKEN_SECRET: ${{ secrets.MODAL_TOKEN_SECRET }}
```
Comment (sambarnes) [inline .github/workflows/release-prod.yml:14]: will need to configure some new secrets in the repo settings, I don't have the perms rn

File: .github/workflows/release-prod.yml (hunk 1 above)
Comment (louisgv) [reply .github/workflows/release-prod.yml:6]: We should add the poetry lock file as well

File: .github/workflows/release-prod.yml (hunk 2 above)
Comment (louisgv) [reply .github/workflows/release-prod.yml:14]: Done! cc @alexanderatallah to config @sambarnes 's perm

**Review by louisgv — APPROVED** [review]
LGTM 👍

**Pull request merged**